### PR TITLE
Added sine calibration blockette detection in the record analyzer

### PR DESCRIPTION
**Pull request by aringler-usgs**
### What does this PR do?

I am adding the ability to detect sine calibration blockettes to the recordanalyzer script.  Assuming this is a reasonable pull-request, that you would be interested in I will add detection of blockettes 310 and 320 as well.

### Why was it initiated?  Any relevant Issues?

I will have a bit of additional time to make contributions in the future.  I would like to add some comparisons to inventory objects.  However, I figured I should do a few smaller PRs before I tackle something big (like comparing inventory objects).  I also wanted to start adding the pieces from https://github.com/aringler-usgs/obspytools/issues/1

I do not see why so many files appear *changed*.  I only changed these three files:
obspy/io/mseed/tests/data/blockette300.mseed
obspy/io/mseed/scripts/recordanalyzer.py
obspy/io/mseed/tests/test_recordanalyzer.py

I would appreciate any idea on why so many files changed.  It looks like some sort of line ending issue.  Did I mess something up?  Thanks for your help.

### PR Checklist
- [X] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [X] This PR is not directly related to an existing issue (which has no PR yet).
- [ ] If the PR is making changes to documentation, docs pages can be built automatically.
      Just remove the space in the following string after the + sign: "+ DOCS"
- [ ] If any network modules should be tested for the PR, add them as a comma separated list
      (e.g. `clients.fdsn,clients.arclink`) after the colon in the following magic string: "+TESTS:"
      (you can also add "ALL" to just simply run all tests across all modules)
- [X] All tests still pass.
- [X] Any new features or fixed regressions are be covered via new tests.
- [X] Any new or changed features have are fully documented.
- [ ] Significant changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

**Comment by megies** [comment]
@aringler-usgs getting these additional blockettes into recordanalyzer is definitely a valuable change that we will happily welcome. Also any other blockettes you feel like adding, of course.

Line endings should only be an issue if windows is involved, I think, and also maybe older git versions. You're not on Windows though, right? Maybe git config manual holds something useful, maybe this `autocrlf` setting.. https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration

Also, one thing I always recommend is to selectively add changes to the staging area. On command line use `git add -p` to be prompted about adding each local change individually (dunno how this translates to GUI git tools). This control over commits is essential IMHO.

And finally, it's probably easiest to just redo the commits and only adding those changes you intended, I guess. If you need help with that just let me know.

**Comment by aringler-usgs** [comment]
Okay, I *think* we managed to prevent the line changes.  @jholland-usgs helped me get the files removed.  They appear to be coming from when I clone it.  I am on a linux machine using git version 1.8.3.1.  So I wonder if it is my git version

I don't have test data for blockette 390 or 395 so I didn't add these.  Please tell me what else you would like changed.

Thanks!
Adam

File: obspy/io/mseed/tests/test_recordanalyzer.py [hunk 1]
```diff
@@ -297,6 +297,123 @@ def test_record_with_sine_calibration(self):
...
+        Rolloff: 3DB@10Hz
+
+CALCULATED VALUES
+    Corrected Starttime: 2018-02-13T20:01:45.069538Z
+
+''' % (filename,)  # noqa
+        self.assertEqual(expected, out.stdout.replace("\t", "    "))  # noqa
+
+
+    def test_record_with_random_calibration(self):
```
Comment (megies) [inline obspy/io/mseed/tests/test_recordanalyzer.py]: Only one blank line in between method blocks

**Comment by megies** [comment]
> I am on a linux machine using git version 1.8.3.1. So I wonder if it is my git version

1.8 is a museum piece, to be honest. Released 2012. I am not surprised at all that strange things happened. You definitely want to grab a binary of a more recent release. ;-)

> I don't have test data for blockette 390 or 395 so I didn't add these

Hmm, maybe you could try the FDSN data mailing list to ask around for test data? But anyway, I guess if you don't have data, most likely very few people will have it and even fewer obspy users might ever need those blockettes in obspy mseed recordanalyzer, so you might as well just leave them out.

**Comment by megies** [comment]
I restarted Circle CI so we get the code linted..

**Comment by aringler-usgs** [comment]
> I restarted Circle CI so we get the code linted (it glitched out somehow)..

I think the code should be flake8 now and it looks to be passing on travis.  I don't know what is going on with Circle CI.  If this is something on my end I am happy to try to fix it.

I think there are a few other pull requests I would like to make before bothering with blockettes 390 and 395.

File: obspy/io/mseed/scripts/recordanalyzer.py [hunk 2]
```diff
@@ -262,6 +263,114 @@ def _parse_blockette(self, blkt_type):
...
+            blkt_dict['Calibration Start Time'] = \
+                UTCDateTime(year=int(unpack_values[0]),
+                            julday=int(unpack_values[1]),
+                            hour=int(unpack_values[2]),
+                            minute=int(unpack_values[3]),
+                            second=int(unpack_values[4]),
+                            microsecond=int(unpack_values[5]))
+
+            blkt_dict['Number of Step Calibrations'] = int(unpack_values[7])
+            blkt_dict['Step Duration'] = int(float(unpack_values[8])/10000.)
```
Comment (krischer) [inline obspy/io/mseed/scripts/recordanalyzer.py]: maybe call this "Step Duration in Seconds"? Just so its not confusing for people who use this with the reference manual at hand.

File: obspy/io/mseed/scripts/recordanalyzer.py [hunk 3]
```diff
@@ -262,6 +263,114 @@ def _parse_blockette(self, blkt_type):
...
+            blkt_dict['Interval Duration'] = int(unpack_values[9])
```
Comment (krischer) [inline obspy/io/mseed/scripts/recordanalyzer.py]: If you convert the step duration to seconds, the interval duration should be as well to be consistent.

File: obspy/io/mseed/scripts/recordanalyzer.py [hunk 4]
```diff
@@ -262,6 +263,114 @@ def _parse_blockette(self, blkt_type):
...
+            blkt_dict['Calibration Start Time'] = \
+                UTCDateTime(year=int(unpack_values[0]),
+                            julday=int(unpack_values[1]),
+                            hour=int(unpack_values[2]),
+                            minute=int(unpack_values[3]),
+                            second=int(unpack_values[4]),
+                            microsecond=int(unpack_values[5]))
+
+            blkt_dict['Calibration Duration'] = \
+                int(float(unpack_values[9])/10000.)
```
Comment (krischer) [inline obspy/io/mseed/scripts/recordanalyzer.py]: Same comment as before with the name.

File: obspy/io/mseed/scripts/recordanalyzer.py [hunk 5]
```diff
@@ -262,6 +263,114 @@ def _parse_blockette(self, blkt_type):
...
```
Comment (krischer) [inline obspy/io/mseed/scripts/recordanalyzer.py]: Once again.

**Comment by krischer** [comment]
Great thanks! Good to merge from my point of view if CI passes.

**Comment by krischer** [comment]
Ah one more thing - can you add an entry to the root level "CHANGELOG.txt" file?

**Comment by krischer** [comment]
Thanks! I fixed a typo in the changelog. The rest looks good now.

**Pull request merged**